code/game/objects/effects/spawners/gibspawner.dm

```diff
@@ -35,3 +35,19 @@
 	gibdirections = list(list(NORTH, NORTHEAST, NORTHWEST),list(SOUTH, SOUTHEAST, SOUTHWEST),list(WEST, NORTHWEST, SOUTHWEST),list(EAST, NORTHEAST, SOUTHEAST), alldirs, alldirs)
 	gibamounts[6] = pick(0,1,2)
 	..()
+
+/obj/effect/gibspawner/blood
+	gibtypes = list(/obj/effect/decal/cleanable/blood/gibs/blood,/obj/effect/decal/cleanable/blood/gibs/blood,/obj/effect/decal/cleanable/blood/gibs/drips,/obj/effect/decal/cleanable/blood/gibs/drips)
+	gibamounts = list(1,1,1,1)
+
+/obj/effect/gibspawner/blood/New()
+	gibdirections = list(alldirs, list(), list(WEST, NORTHWEST, SOUTHWEST, NORTH),list(EAST, NORTHEAST, SOUTHEAST, SOUTH))
```

Comment:
Yeah, it took me a lot of effort but after reading and re-reading a million times how the gibs code work, I think I now fully understand it. Basically, each gibtype decal corresponds to a direction that is in the same place as in the gibdirections list. 
So, in action and in order, first decal could go to any direction, second decal stays at wherever `src` is since the list is empty, third and fourth decals go to 1 direction from each list (picked randomly).